shared/ClrFacade/Internal.cs

```diff
@@ -13,17 +13,28 @@ public static class Internal
 {
    public static int CallInstanceMethod(IntPtr obj, string methodName, IntPtr arguments, int numObjects, IntPtr returnValue)
    {
-      var objectArguments = convertToObjectArguments(arguments, numObjects);
-
-      var instPtr = Marshal.ReadIntPtr(obj, 0);
-      var t = Marshal.PtrToStructure<RSharpGenericValue>(instPtr);
-      var instance = convertRSharpParameters(new[] { t })[0];
+      try
+      {
+         var objectArguments = convertToObjectArguments(arguments, numObjects);
 
-      var result = CallInstanceMethod(instance, methodName, objectArguments);
-      var tempRetVal = RSharpGenericValueExtensions.FromObject(result);
+         var instPtr = Marshal.ReadIntPtr(obj, 0);
+         var t = Marshal.PtrToStructure<RSharpGenericValue>(instPtr);
+         var instance = convertRSharpParameters(new[] { t })[0];
 
-      Marshal.StructureToPtr(tempRetVal, returnValue, false);
-      return 1;
+         var result = CallInstanceMethod(instance, methodName, objectArguments);
+         var tempRetVal = RSharpGenericValueExtensions.FromObject(result);
+         Marshal.StructureToPtr(tempRetVal, returnValue, false);
+      }
+      catch (Exception ex)
+      {
+         if (!logThroughR(ex))
```

Comment:
I think we should change how this works for clarity. Functionally it will be the same.

```logThroughR``` always returns false. So, why bother checking it? make the method void and take the if statement away in all these cases.

This makes it look like some decision is being taken, but it's not. We caught an exception and we will return -1 and there will be a string returned in ```returnValue```.

Reply:
In fact, ```logThroughR``` could return a tuple with the exception and string and we would not need the static ```LastCallException```.  I'm pretty sure ```LastException``` is totally unused. Unless it's somehow checked dynamically. I don't think so though because ```Internal``` is completely new class so it would not be accessed directly by R code.

Reply:
there is a function in R thats called printTraceback that uses/gets the LastException. I believe this was how we dealt with exceptions in the old code. I kept it as is so that it can still be used, just in case.

Reply:
`LastException` is used by the R function `printTraceback()` which prints the last exception. The reason is that R truncates error messages, so sometime it could be useful to call it.

`Internal` is not really new, its contents were in `ClrFacade` and yes, it broke some R code and I had to use the new assembly name ;)

Reply:
So you want to keep this printTraceback()? If so, then the LastException or whatever should be moved, or proxied in ClrFacade instead.

Reply:
Also add tests to make sure the proxies stay working.